src/test/java/org/gridsuite/modification/server/utils/assertions/AbstractModificationInfosAssert.java

```diff
@@ -0,0 +1,48 @@
+package org.gridsuite.modification.server.utils.assertions;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.recursive.comparison.RecursiveComparisonConfiguration;
+import org.gridsuite.modification.server.dto.ModificationInfos;
+
+import java.time.temporal.Temporal;
+import java.util.Objects;
+import java.util.UUID;
+
+public abstract class AbstractModificationInfosAssert<SELF extends AbstractModificationInfosAssert<SELF, ACTUAL>, ACTUAL extends ModificationInfos>
+    extends AbstractAssert<SELF, ACTUAL> {
+    protected AbstractModificationInfosAssert(ACTUAL actual, Class<SELF> assertClass) {
+        super(actual, assertClass);
+    }
+
+    public SELF hasUuid(UUID uuid) {
+        isNotNull();
+        if (!Objects.equals(actual.getUuid(), uuid)) {
+            failWithMessage("Expected Modification UUID to be <%s> but was <%s>", uuid, actual.getUuid());
+        }
+        return myself;
+    }
+
+    public SELF hasDate(Temporal dateTime) {
+        isNotNull();
+        Assertions.assertThat(actual.getDate())
+                  .withFailMessage("Expected Modification date-time to be <%s> but was <%s>", dateTime, actual.getDate())
+                  .isEqualTo(dateTime);
+        return myself;
+    }
+
+    protected RecursiveComparisonConfiguration recursiveConfiguration() {
+        return RecursiveComparisonConfiguration.builder()
+            .withIgnoreAllOverriddenEquals(true)
```

Comment:
Why ?

Reply:
because I see some equals don't compare all fields, and we compare all fields while ignoring some which don't follow same rules than `equals()`

Reply:
Ah ok